testcontainers/src/core/image/runnable_image.rs

```diff
@@ -38,6 +41,33 @@ pub enum Host {
     HostGateway,
 }
 
+#[derive(Debug, Clone, Copy)]
+pub enum CgroupnsMode {
+    Host,
+    Private,
+}
+
+impl TryFrom<HostConfigCgroupnsModeEnum> for CgroupnsMode {
+    type Error = ();
+
+    fn try_from(value: HostConfigCgroupnsModeEnum) -> Result<Self, Self::Error> {
+        match value {
+            HostConfigCgroupnsModeEnum::EMPTY => Err(()),
+            HostConfigCgroupnsModeEnum::PRIVATE => Ok(Self::Private),
+            HostConfigCgroupnsModeEnum::HOST => Ok(Self::Host),
+        }
+    }
+}
```

Comment:
Do we really need this conversion? As far as I can see we need to convert `CgroupnsMode` to `HostConfigCgroupnsModeEnum` and not vice-versa 

Additionally, since this conversion is needed for `async_runner` - I would suggest moving the implementation there

Reply:
Hi @DDtKey, thanks for the feedback
You are right, I removed the unused TryFrom and added some tests as you suggested.